src/components/button/Button.tsx

```diff
@@ -22,33 +21,58 @@ const textComponentMap = {
   large: UITextDisplay,
 };
 
-/** @type {React.ComponentType<any>} */
+interface Props {
+  /** The content to display inside the button. */
+  children?: ReactNode;
+  /** A class name for the button to give custom styles. */
+  className?: string;
+  /** The color which the button should have when 'level' is set to 'outline' */
+  color?: 'teal' | 'neutral' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'white';
+  /** A custom element to be rendered */
+  element?: any;
+  /** If true, component will be disabled. */
+  disabled?: boolean;
+  /** If true, component will be shown in an active state */
+  active?: boolean;
+  /** If true, component will take the full width available. */
+  fullWidth?: boolean;
+  /** The icon displayed inside the button. */
+  icon?: ReactNode;
+  /** The position of the icon inside the button. */
+  iconPlacement?: 'left' | 'right';
+  /** The textual label displayed inside the button. */
+  label?: string;
+  /** Determines which kind of button to be rendered. */
+  level?: BUTTON_LEVELS;
```

Comment:
You're gonna have to "cast' it to a type, otherwise you can't use ith with string literals (`<Button level="primary" />` is going to have an error).